### ISSUE TITLE
rand_tangent(adjoint(array)) returns Composite even though Adjoint is a natural differential

### ISSUE THREAD

**Issue by mzgubic** [JuliaDiff/ChainRulesTestUtils.jl#130]
Presently
```julia
julia> A = rand(3, 3);

julia> adj = adjoint(A);

julia> ChainRulesTestUtils.rand_tangent(adj)
Composite{Adjoint{Float64,Array{Float64,2}}}(parent = [-0.5319309079526605 -1.2397774126992458 1.450107932767501; 0.2443697244819896 -0.5283784206385908 -0.36484818100635125; -0.18829408436246423 -1.0652797084400936 -1.0457119567770545],)
```

whereas we would like to get an Adjoint out, which is a natural differential type.

Probably just need to define a new dispatch for `rand_tangent`?

**Comment by oxinabox** [comment]
if we did #123 it would cover this implictly.

**Comment by willtebbutt** [comment]
> Probably just need to define a new dispatch for rand_tangent?

Yes, this makes sense. Alternatively, we could just add a `kwarg` that says whether to produce a natural differential or not.

**Comment by oxinabox** [comment]
> Alternatively, we could just add a kwarg that says whether to produce a natural differential or not.

Can't really do that because `rand_tangent` is mostly used implictity when the user doesn't provide a differential.
We need to chose one.
And a natural differential is the right choice if it is available.

**Comment by willtebbutt** [comment]
> Can't really do that because rand_tangent is mostly used implictity when the user doesn't provide a differential.

Fair point.

>  And a natural differential is the right choice if it is available.

Is it? What about functions that you can AD automatically that involve `Adjoint{Float64, Matrix{Float64}}`? They'll produce composites.

**Comment by oxinabox** [comment]
I agree it isn't the only choice.
But I do claim it is the right one.
the reason we call it natural is it is the one humans use when writing rules.
This package is (primarily) for testing rules humans write.
and it works best if you don't mix natural and structural deriviatives
(we do need to solve that, but its not a problem for this package)
